src/deck.ts

```diff
@@ -0,0 +1,90 @@
+type Suit = "spades" | "hearts" | "diamonds" | "clubs"
+
+type Rank = 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14
+
+type ordering = 1 | 0 | -1
+            // GT | EQ| LT
+
+interface Card {
+    rank: Rank
+    suit: Suit
+}
+
+function cardCompare(card1: Card, card2: Card): ordering {
+    if (card1.rank > card2.rank) {
+        return 1
+    } else if (card1.rank < card2.rank) {
+        return -1
+    } else return 0
+}
+
+function cardCompareTrump(card1: Card, card2: Card, trump: Suit = "spades"): ordering {
+    if (card1.suit == trump) {
+        return 1
+    } else if (card2.suit == trump) {
+        return -1
+    } else return cardCompare(card1, card2)
+}
+
+
+class Deck {
+    deck: Card[]
+    hands: Card[][]
+    trumpCard: Card
```

Comment:
Is there a reason we need to store the trump card? it seems like once we've figured out trump, there's no reason to store this

Reply:
because it's taken out of the deck and is always the last card drawn, this makes it easier on the draw function.

Reply:
going to leave this here though.